version-migration/v0.6-to-v0.7-migration/migrated-examples/build.gradle

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at:
+ *
+ *       http://aws.amazon.com/apache2.0/
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific
+ *  language governing permissions and limitations under the License.
+ */
+
+plugins {
+    id 'java-library'
+}
+
+dependencies {
+    implementation("org.jetbrains.kotlin:kotlin-stdlib-jdk8")
+
+    testImplementation("org.jetbrains.kotlin:kotlin-test")
+    testImplementation("org.junit.jupiter:junit-jupiter-params:5.8.2")
+
+    testImplementation("org.partiql:partiql-lang-kotlin:0.7.0")
+}
+
+java {
+    sourceCompatibility = JavaVersion.VERSION_11
+    targetCompatibility = JavaVersion.VERSION_11
+}
```

Comment:
Not too sure why but for v0.7.0 only (previous version and v0.8.0 work fine), I needed to specify the `sourceCompatability` and `targetCompatability`. Without this I got the following error locally and in GitHub Actions:

```
Execution failed for task ':version-migration:v0.6-to-v0.7-migration:migrated-examples:compileTestKotlin'.
> Error while evaluating property 'filteredArgumentsMap' of task ':version-migration:v0.6-to-v0.7-migration:migrated-examples:compileTestKotlin'
   > Could not resolve all files for configuration ':version-migration:v0.6-to-v0.7-migration:migrated-examples:testCompileClasspath'.
      > Could not resolve org.partiql:partiql-lang-kotlin:0.7.0.
        Required by:
            project :version-migration:v0.6-to-v0.7-migration:migrated-examples
         > No matching variant of org.partiql:partiql-lang-kotlin:0.7.0 was found. The consumer was configured to find an API of a library compatible with Java 8, preferably in the form of class files, preferably optimized for standard JVMs, and its dependencies declared externally, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'jvm' but:
             - Variant 'apiElements' capability org.partiql:partiql-lang-kotlin:0.7.0 declares an API of a library, packaged as a jar, and its dependencies declared externally, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'jvm':
                 - Incompatible because this component declares a component compatible with Java 11 and the consumer needed a component compatible with Java 8
                 - Other compatible attribute:
                     - Doesn't say anything about its target Java environment (preferred optimized for standard JVMs)
             - Variant 'runtimeElements' capability org.partiql:partiql-lang-kotlin:0.7.0 declares a runtime of a library, packaged as a jar, and its dependencies declared externally, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'jvm':
                 - Incompatible because this component declares a component compatible with Java 11 and the consumer needed a component compatible with Java 8
                 - Other compatible attribute:
                     - Doesn't say anything about its target Java environment (preferred optimized for standard JVMs)
```

Perhaps there was some conflict with the compatibility set in the root `build.gradle` file.

Reply:
Does it work when you set it to JavaVersion.VERSION_8?

Reply:
No, still get the same error:

```
* What went wrong:
Execution failed for task ':version-migration:v0.6-to-v0.7-migration:migrated-examples:compileTestKotlin'.
> Error while evaluating property 'filteredArgumentsMap' of task ':version-migration:v0.6-to-v0.7-migration:migrated-examples:compileTestKotlin'
   > Could not resolve all files for configuration ':version-migration:v0.6-to-v0.7-migration:migrated-examples:testCompileClasspath'.
      > Could not resolve org.partiql:partiql-lang-kotlin:0.7.0.
        Required by:
            project :version-migration:v0.6-to-v0.7-migration:migrated-examples
         > No matching variant of org.partiql:partiql-lang-kotlin:0.7.0 was found. The consumer was configured to find an API of a library compatible with Java 8, preferably in the form of class files, preferably optimized for standard JVMs, and its dependencies declared externally, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'jvm' but:
             - Variant 'apiElements' capability org.partiql:partiql-lang-kotlin:0.7.0 declares an API of a library, packaged as a jar, and its dependencies declared externally, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'jvm':
                 - Incompatible because this component declares a component compatible with Java 11 and the consumer needed a component compatible with Java 8
                 - Other compatible attribute:
                     - Doesn't say anything about its target Java environment (preferred optimized for standard JVMs)
             - Variant 'runtimeElements' capability org.partiql:partiql-lang-kotlin:0.7.0 declares a runtime of a library, packaged as a jar, and its dependencies declared externally, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'jvm':
                 - Incompatible because this component declares a component compatible with Java 11 and the consumer needed a component compatible with Java 8
                 - Other compatible attribute:
                     - Doesn't say anything about its target Java environment (preferred optimized for standard JVMs)
```

Reply:
How odd. I have a feeling the snipped introduced in the root `build.gradle`:
```groovy
plugins.withId('java') {
  // compatibility specification
}
```

isn't working. Do you want to open an issue so someone can take a look into it?

Reply:
Sure, I can cut an issue to track.

Reply:
Created https://github.com/partiql/partiql-lang-kotlin/issues/858.